gammapy/utils/random.py

```diff
@@ -30,6 +30,14 @@ def sample_sphere(size, lon_range=None, lat_range=None, unit='radians'):
     lon, lat: arrays
         Longitude and latitude coordinate arrays
     """
+    # Correctly interpret units at an early stage
+    if unit in ['rad', 'radian', 'radians']:
```

Comment:
We should use the `astropy.coordinates.Longitude` and `astropy.coordinates.Latitude` classes, which take care of unit handling. Can you please rewrite this function (in a separate PR or here, as you like) to use the Astropy classes instead? 

There's probably a few callers to this function which you can find e.g. via the full-text search on Github. Please fix these also.

PS: quite a bit of code in Gammapy was written before `Quantity` existed, i.e. needs to be updated to use quantities where appropriate.